### PR TITLE
Partition explainer PyTorch support

### PR DESCRIPTION
**Pull request by alexander-pv**
Hi, thanks for your work!

This is the updated #2400 PR.

Summary:
1. Partition explainer supports torch classification models.
2. New Jupyter notebook example with PyTorch.

### CODE REVIEW

**Comment by slundberg** [comment]
Thanks! Sorry I was out for a bit (after paternity leave), will give this a review once I re-run the CI tests.

File: shap/plots/_image.py [hunk 1]
```diff
@@ -93,60 +116,69 @@ def image(shap_values, pixel_values=None, labels=None, width=20, aspect=0.2, hsp
...
+            axes[row, i + 1].axis('off')
     if hspace == 'auto':
         fig.tight_layout()
     else:
         fig.subplots_adjust(hspace=hspace)
-    cb = fig.colorbar(im, ax=np.ravel(axes).tolist(), label="SHAP value", orientation="horizontal", aspect=fig_size[0]/aspect)
+    cb = fig.colorbar(im, ax=np.ravel(axes).tolist(), label="SHAP value", orientation="horizontal",
+                      aspect=fig_size[0] / aspect)
     cb.outline.set_visible(False)
+    if save_name:
```
Comment (slundberg) [inline shap/plots/_image.py]: In all the other plotting functions the expected way to save plots is to just call `pl.savefig` after calling the shap plot function. So I am inclined to remove this for consistency. Any comments on that?

**Comment by slundberg** [comment]
Thanks again! This looks good. See my one comment on the `save_name` option.

File: shap/plots/_image.py (hunk 1 above)
Comment (alexander-pv) [reply shap/plots/_image.py]: Hi! Sorry I missed that. That's ok.

Comment (slundberg) [reply shap/plots/_image.py]: No problem. it should probably get noted in the docs more clearly. I'll delete then merge once the tests pass

**Comment by alexander-pv** [comment]
Hi, thanks for the answer and the review!

**Pull request merged**